doc/api/dns.md

```diff
@@ -173,15 +173,13 @@ Valid values for `rrtype` are:
  * `'SOA'` - start of authority record
  * `'NAPTR'` - name authority pointer record
 
-The `callback` function has arguments `(err, addresses)`. When successful,
-`addresses` will be an array, except when resolving an SOA record which returns
-an object structured in the same manner as one returned by the
-[`dns.resolveSoa()`][] method. The type of each item in `addresses` is
-determined by the record type, and described in the documentation for the
-corresponding lookup methods.
+The `callback` function has arguments `(err, addresses)` and will receive an
+array of results. A result will be a string except when `rrtype` is `MX` or
+`SOA` where it will be an object structured like the results of
+[`dns.resolveMX()`][] and [`dns.resolveSoa()`][], respectively.
 
-On error, `err` is an [`Error`][] object, where `err.code` is
-one of the error codes listed [here](#dns_error_codes).
+On error, `err` is an [`Error`][] object, where `err.code` is one of the error
+codes listed [here](#dns_error_codes).
```

Comment:
Nit: Avoid `here` as link text. Maybe `...one of the error codes listed under [DNS error codes].`?